backend/app/services/gmail_service.py

```
@@ -0,0 +1,431 @@
+                return category
+        
+        return 'newsletter'
+    
+    def _detect_language(self, text: str) -> str:
+        """テキストの言語を検出"""
+        try:
+            from langdetect import detect
+            lang = detect(text[:1000])  # 最初の1000文字で判定
+            if lang in ['ja', 'en', 'zh']:
+                return lang
+        except:
```

Ruff (0.12.2)

Do not use bare `except`

(E722)